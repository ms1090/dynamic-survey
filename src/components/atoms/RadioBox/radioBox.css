.radio-box {
  background: #f8f9fb;
  border: solid 1px #8c8c8a;
  border-radius: 3px;
  color: #5d5d5d;
  cursor: pointer;
  display: block;
  margin-bottom: 8px;
  padding: 10px 16px;
}