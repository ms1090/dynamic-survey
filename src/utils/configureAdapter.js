import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
