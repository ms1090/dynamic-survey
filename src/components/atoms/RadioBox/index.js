import React from 'react';
import PropTypes from 'prop-types';
import './radioBox.css';

/**
 * Display a radio button as per customization props
 * @param {String} className CSS class name
 * @param {String} id Input element id
 * @param {String} name Input element name
 * @param {String} label Label for input element
 * @return {React.Element} Return radio button UI
 */
const RadioBox = ({
  className,
  id,
  name,
  label,
  ...others
}) => (
  <label className={`radio-box ${className}`} htmlFor={id}>
    <input
      type="radio"
      id={id}
      name={name}
      value={label}
      {...others}
    />
    <span>{label}</span>
  </label>
);
// default values for component props
RadioBox.defaultProps = {
  className: '',
  id: '',
  name: '',
  label: '',
};
// prop types checking
RadioBox.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.string,
};
export default RadioBox;
