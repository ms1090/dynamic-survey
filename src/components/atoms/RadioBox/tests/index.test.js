import React from 'react';
import { shallow } from 'enzyme';
import RadioBox from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('RadioBox Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<RadioBox
      id="testId"
      className="test-className"
      name="test-name"
      label="test label"
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
