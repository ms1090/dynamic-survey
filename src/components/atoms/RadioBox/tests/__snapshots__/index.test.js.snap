// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioBox Component should render correctly 1`] = `
<label
  className="radio-box test-className"
  htmlFor="testId"
>
  <input
    id="testId"
    name="test-name"
    type="radio"
    value="test label"
  />
  <span>
    test label
  </span>
</label>
`;
